stackstate_checks_base/tests/test_http_helper.py

```diff
@@ -0,0 +1,458 @@
+            assert False
+        except ValueError:
+            assert True
+
+    def test_body_type_validation(self):
+        http = HTTPHelper()
+        valid_response = http.get({
+            "endpoint": "mock://test.com",
+            "mock_enable": True,
+            "mock_status": 200,
+            "mock_response": {
+                'hello': 'world'
+            },
+        })
+        invalid_response = http.get({
+            "endpoint": "mock://test.com",
+            "mock_enable": True,
+            "mock_status": 200,
+            "mock_response": "test",
+        })
+
+        assert HTTPHelperResponseHandler().validate_body_type() is True
+        assert HTTPHelperResponseHandler({"response_type_validation": HTTPRequestType.JSON}) \
+               .get_response_model().response_type_validation == HTTPRequestType.JSON
+        assert HTTPHelperResponseHandler({"response_type_validation": -1}) \
+               .get_response_model().response_type_validation == -1
+        assert HTTPHelperResponseHandler({
+            "response_type_validation": HTTPRequestType.JSON
+        }).validate_body_type(valid_response)
+        try:
+            HTTPHelperResponseHandler({
+                "response_type_validation": HTTPRequestType.JSON
+            }).validate_body_type(invalid_response)
+            assert False
+        except ValueError:
+            assert True
+
+
+class TestHTTPHelperBase(unittest.TestCase):
+    def test_full_get(self):
+        http = HTTPHelper()
+        response = http.get({
+            "endpoint": "mock://test.com",
+            "mock_enable": True,
+            "mock_status": 200,
+            "mock_response": {
+                'hello': 'world',
+                'pong': True,
+            },
+            "auth_data": {
+                'username': 'world',
+                'password': 'hello'
+            },
+            "auth_type": HTTPRequestType.JSON,
+            "session_auth_data": {
+                'username': 'world',
+                'password': 'hello'
+            },
+            "session_auth_type": HTTPRequestType.JSON,
+            "body": {"hello": "world", "pong": True},
+            "proxy": {"hello": "world"},
+            "headers": {"a": "1"},
+            "session_headers": {"b": "1"},
+            "query": {"c": "1"},
+            "timeout": 30,
+            "ssl_verify": True,
+            "retry_policy": {},
+            "request_schematic_validation": BodyRequestSchematicTest,
+            "request_type_validation": HTTPRequestType.JSON,
+            "response_status_code_validation": 200,
+            "response_type_validation": HTTPResponseType.JSON,
+            "response_schematic_validation": BodyResponseSchematicTest,
+        })
+
+        assert response.request.method == "GET"
+        assert response.status_code == 200
+        assert json.loads(response.content.decode(response.encoding)).get("hello") == "world"
```

Comment:
I would expect `json.loads(response.content.decode(response.encoding)` this code to be taken care of by the response handler. You are now leaking the `requests` library to the user, which you actually meant to wrap. This now forms a leaky abstraction. 

My advice: do not provide the actual response back to the user, but a wrapper around it that you control. You can simply have two methods on that wrapper like `get_json` and `get_text`. 

You can even call the validate on the response only when the user calls `response.get_json()` and not trigger it on `response.get_text()`.